source/fuzz/fuzzer_pass_add_dead_blocks.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/fuzzer_pass_add_dead_blocks.h"
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/transformation_add_dead_block.h"
+
+namespace spvtools {
+namespace fuzz {
+
+FuzzerPassAddDeadBlocks::FuzzerPassAddDeadBlocks(
+    opt::IRContext* ir_context, FactManager* fact_manager,
+    FuzzerContext* fuzzer_context,
+    protobufs::TransformationSequence* transformations)
+    : FuzzerPass(ir_context, fact_manager, fuzzer_context, transformations) {}
+
+FuzzerPassAddDeadBlocks::~FuzzerPassAddDeadBlocks() = default;
+
+void FuzzerPassAddDeadBlocks::Apply() {
+  std::vector<opt::BasicBlock*> candidate_blocks;
+  for (auto& function : *GetIRContext()->module()) {
+    for (auto& block : function) {
+      if (!GetFuzzerContext()->ChoosePercentage(
+              GetFuzzerContext()->GetChanceOfAddingDeadBlock())) {
+        continue;
+      }
+      if (block.IsLoopHeader()) {
```

Comment:
Bit of a nit, but these checks are repeated in `IsApplicable` (with comments!). Maybe you could create the transformation (the `condition` doesn't matter at this stage; `fresh_id` would have to pass the check somehow...not ideal), check if it `IsApplicable`, then fix up the transformation with a randomly chosen condition and with a fresh id. OK, maybe this is more confusing. Could extract the common checks, but again maybe this is more confusing than a bit of repetition.

Reply:
Actually I think you're right - we don't mind wasting some fresh ids, so I have written the code now so that it populates a list of transformations (speculatively), and then goes ahead and applies the ones that turn out to be applicable.  That indeed eliminates all this re-checking of applicability.  I have done that for some other fuzzer passes, and I think it's a good plan unless checking full applicability is particularly expensive.